client/my-sites/stats/controller.jsx

```diff
@@ -472,11 +473,12 @@ export default {
 		const state = context.store.getState();
 		const siteId = getSelectedSiteId( state );
 		const isJetpack = isJetpackSite( state, siteId );
+		const isPaidPlan = isSiteOnPaidPlan( state, siteId );
 		const startDate = i18n.moment( context.query.startDate, 'YYYY-MM-DD' ).isValid()
 			? context.query.startDate
 			: undefined;
 
-		if ( siteId && ! isJetpack ) {
+		if ( ( siteId && ! isJetpack ) || ! isPaidPlan ) {
```

Comment:
I'm pretty sure there's a race in here that will redirect sites with a paid plan if we load this page before the sites have loaded.

did you check with `force-sympathy`? try loading the activity URL and see if it redirects even though it shouldn't

cc: @nylen

Reply:
@rcoll we could alternatively pass in the route action to the component so that it can make its determination when it wants. we can hide the implementation behind a React interface…

```js
const props = {
	…
	onNotAvailable: () => page.redirect( '/stats' );
}
```

Now we can always display the component. It will load, the sites will load. We can optimize for the case where the sites _are_ loaded

```js
if ( (siteId && ! isJetpack) || (siteId && ! isPaidPlan) ) {
	page.redirect( '/stats' );
} else {
	// as before but also pass in `onNotAvailable`
}